### PR TITLE
Binding to Child View's BindingContextProperty

### PR DESCRIPTION
**Pull request by borbmizzet**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Adds a demo to the playground for binding a child view's BindingContextProperty to a parent view model's member.

### :arrow_heading_down: What is the current behavior?
Xaml binding a child view's BindingContextProperty no longer properly synchronizes between parent view model and child view.

### :boom: Does this PR introduce a breaking change?
no

### :bug: Recommendations for testing
Playground.Forms.Droid ->Show Content View -> Tap "Change Child2 Text"

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [x] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by borbmizzet** [comment]
Hold on, there is a typo in my demo that I'm about to push a correction for.

File: Projects/Playground/Playground.Forms.UI/Pages/ParentContentPage.xaml [hunk 1]
```diff
@@ -8,9 +8,11 @@
              x:Name="ParentPage">
     <ContentPage.Content>
         <StackLayout>
-            <views:ChildContentView ViewModel="{Binding ChildViewModel1}" IsEnabled="{Binding ViewModel.ChildViewModelEnabled, Source={x:Reference ParentPage}}"/>
-            <Button Text="Change Child Text" Command="{Binding ChangeButtonCmd}" />
-            <Button Text="Toggle Child Enabled" Command="{Binding ToggleChildEnabledCmd}" />
+            <views:ChildContentView ViewModel="{Binding Source={x:Reference ParentPage}, Path=ViewModel.ChildViewModel1}" IsEnabled="{Binding ViewModel.ChildViewModelEnabled, Source={x:Reference ParentPage}}"/>
```
Comment (nickrandolph) [inline Projects/Playground/Playground.Forms.UI/Pages/ParentContentPage.xaml]: Is there a reason why "Source" needs to be specified for either of these databinding expressions?

Comment (borbmizzet) [reply Projects/Playground/Playground.Forms.UI/Pages/ParentContentPage.xaml]: Not in this specific context, but if Source is not specified then the binding works a grand total of one time, after which the child view's base.BindingContext is no longer the same as the parent view's ViewModel and it wouldn't be able to find the property ChildViewModel1 in the class 'ChildViewModel'.

**Comment by nickrandolph** [comment]
@borbmizzet can you check over my changes and confirm that it fixes the issue with the binding? I've changed the binding expressions in the page to what they should be - there shouldn't be a need to reference the page as the source for all the expressions.

**Comment by nickrandolph** [comment]
@martijn00 can you review my changes. Was there a reason for the original code?

**Pull request merged**